### ISSUE TITLE
Feature request: Support NoFieldSelectors

### ISSUE THREAD

**Issue by cjay** [ndmitchell/hlint#1401]
Not supported in 3.3.6, and couldn't find a trace of it in the repo, so I think it still is unsupported.

**Comment by ndmitchell** [comment]
Generally, extensions like NoFieldSelectors don't need support, because they don't impact HLint at the parsing level it works at. Or are you seeing an error if you pass the extension? Or something else?

**Comment by cjay** [comment]
v3.3.6 reports `Error: Parse error: Unsupported extension: NoFieldSelectors` on a file that has `{-# LANGUAGE NoFieldSelectors #-}` at the top. Trying to pass it as an argument also fails: `hlint -XNoFieldSelectors <file>` results in `hlint: Unknown extension: 'NoFieldSelectors'`.

**Comment by ndmitchell** [comment]
In #1094 we made it an error if an extension is not recognised. GHC 9.2.1 introduced FieldSelectors and NoFieldSelectors, but it seems we didn't pick them up. @shayne-fletcher - my guess is this is a bug in `Language.Haskell.GhclibParserEx.GHC.Driver.Session.readExtension`? But not sure how it is missing.

**Comment by shayne-fletcher** [comment]
@ndmitchell @cjay `FieldSelectors` was introduced in ghc-9.2.1. hlint-3.3.6 uses the ghc-9.0 parse tree. it wasn't until hlint-3.4 that it began to use the 9.2 parse tree (https://github.com/ndmitchell/hlint/pull/1247).

indeed, for the program,
```haskell
{-# LANGUAGE NoFieldSelectors #-}
main = undefined
```
using [hlint-3.3.6](https://hackage.haskell.org/package/hlint-3.3.6/hlint-3.3.6.tar.gz) i get
```
NoFieldSelectors.hs:1:1: Error: Parse error: Unsupported extension: NoFieldSelectors
```
whereas using [hlint-3.4](https://hackage.haskell.org/package/hlint-3.4/hlint-3.4.tar.gz) (and a build of hlint from `HEAD`) i observe,
```
NoFieldSelectors.hs:1:1-33: Warning: Avoid restricted extensions
Found:
  {-# LANGUAGE NoFieldSelectors #-}
Note: may break the code

1 hint
```

**Comment by ndmitchell** [comment]
Thanks @shayne-fletcher - so @cjay, seems like fixed in the latest version.

**Comment by cjay** [comment]
Thanks everyone :)
(not sure if the issue should stay open because of the warning, so not closing it myself)

**Comment by ndmitchell** [comment]
The warning only occurs in HLint itself, because it has a .hlint.yaml saying all non-approved extensions must be listed explicitly. Won't happen in your code, so closing.
